src/sentry/deletions/tasks/groups.py

```diff
@@ -47,6 +47,8 @@ def delete_groups(
     if not first_group:
         raise DeleteAborted("delete_groups.no_group_found")
 
+    transaction_id = transaction_id or uuid4().hex
```

Comment:
This guarantees that the `set_tags()` call below will not be `None`.

Reply:
Thanks for noticing it!
It seems that it was only possible through the tests.
I have tightened the signature and removed this.